.pre-commit-config.yaml

```diff
@@ -85,7 +85,7 @@ repos:
         additional_dependencies:
           - aiohttp
           - coredis
-          - fhaviary[llm]>=0.8.2 # Match pyproject.toml
+          - fhaviary[llm]>=0.10.0 # Match pyproject.toml
```

Comment:
A few line below, let's remove the:

```toml
- openai>=1 # Match pyproject.toml
```

Reply:
Do you not want to have openai type checking?

Reply:
So `mypy` will still get `openai` through `litellm`, as `openai` is a `litellm` dependency. So we'll still have `openai` there.

The idea is we want the `mypy` list to match our first party deps in `dependencies`.

We made `openai` a first party dependency temporarily just to ensure its version matched. So we had to do the same in `mypy`. Now that this PR removes it from our first party deps, we want to remove it from our `mypy` list as well to keep them in sync

Reply:
Got it - this section of mypy has many inconsistencies with the pyproject.toml file. Will fix it I guess

Reply:
Like `litellm` isn't even in the `mypy` file

Reply:
Yes you are dead on 🥳 

One last dimension (we also just talked in person) is that some packages are untyped: https://github.com/Future-House/paper-qa/blob/v5.3.2/pyproject.toml#L144-L149, so we don't need to give them to `mypy`'s isolated list here. Respecifying them (even though they aren't typed):
- Pro: future looking for if they become typed
- Con: bloats `pre-commit`'s isolated venv a bit

That being said, it's an easier heuristic to just "exact match" dependency lists. So let's leave as is, thanks and resolving